lib/kafo/progress_bar.rb

```diff
@@ -42,15 +42,28 @@ def update(line)
       end
 
       if (line_start = EVALTRACE_START.match(line))
-        if (known_resource = find_known_resource(line_start[1]))
+        if line_start[4]
+          # Puppet 6.6 introduced progress in evaltrace
+          # Puppet counts 1-based where we count 0-based here
+          new_lines = line_start[3].to_i - 1
+          new_total = line_start[4].to_i
+          if new_lines != @lines || @total != new_total
+            @lines = new_lines
+            @total = new_total
+            update_bar = true
+            force_update = true
+          end
+        end
+
+        if (known_resource = find_resource(line_start[1]))
           line = known_resource
           update_bar = true
           force_update = true
         end
       end
 
       if (line_end = EVALTRACE_END.match(line)) && @total != :unknown && @lines < @total
-        if (known_resource = find_known_resource(line_end[1]))
+        if (known_resource = find_resource(line_end[1]))
           @resources.delete(known_resource)  # ensure it's only counted once
           @lines += 1
         end
```

Comment:
This feels wrong. It will always find something. I think that means it always increases `@lines` by 1, introducing an off-by-1 error.

Reply:
Why will this always find something? If find_resource returns nil this won't trigger. I've not seen any issues testing this in practice.